From 53bf494b500a7362ce83fa0bac6068bc7a511c4e Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 1 Sep 2014 20:15:20 -0400
Subject: [powerpc] Add hvcalls for 24x7 and gpci (Get Performance Counter Info)

Message-id: <1409602534-23951-2-git-send-email-gduarte@redhat.com>
Patchwork-id: 90114
O-Subject: [RHEL7.1 PATCH BZ 947159 01/15] powerpc: Add hvcalls for 24x7 and gpci (Get Performance Counter Info)
Bugzilla: 947159
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=947159
Upstream Status: 827f798ac13c27c4a122175c60a3a684b59e9103

commit 827f798ac13c27c4a122175c60a3a684b59e9103
Author: Cody P Schafer <cody@linux.vnet.ibm.com>
Date:   Fri Mar 14 16:00:36 2014 +1100

    powerpc: Add hvcalls for 24x7 and gpci (Get Performance Counter Info)

    Signed-off-by: Cody P Schafer <cody@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/hvcall.h b/arch/powerpc/include/asm/hvcall.h
index 0c7f2bf..bade5ef 100644
--- a/arch/powerpc/include/asm/hvcall.h
+++ b/arch/powerpc/include/asm/hvcall.h
@@ -274,6 +274,11 @@
 /* Platform specific hcalls, used by KVM */
 #define H_RTAS   0xf000
 
+/* "Platform specific hcalls", provided by PHYP */
+#define H_GET_24X7_CATALOG_PAGE 0xF078
+#define H_GET_24X7_DATA  0xF07C
+#define H_GET_PERF_COUNTER_INFO 0xF080
+
 #ifndef __ASSEMBLY__
 
 /**
-- 
1.7.1